core/unit_test/TestIssue5594.hpp

```diff
@@ -0,0 +1,84 @@
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+namespace TestViewSubview {
+
+namespace issue_5594 {
+template <typename View>
+View create_view() {
+  using execution_space = typename View::device_type::execution_space;
+  View view("", 10);
+  Kokkos::parallel_for(
+      "test_view_subview_const_randomaccess",
+      Kokkos::RangePolicy<execution_space>(0, view.extent(0)),
+      KOKKOS_LAMBDA(const int i) { view(i) = i; });
+  return view;
+}
+}  // namespace issue_5594
+
+/* Test creating a RandomAccess subview of a view of const T
+ */
+template <typename Space>
+void test_view_subview_const_randomaccess() {
+  using execution_space = typename Space::execution_space;
+  using view_t          = Kokkos::View<int*, Space>;
+  using view_const_t    = Kokkos::View<const int*, Space>;
+  using um_view_const_t = Kokkos::View<
+      const int*, Space,
+      Kokkos::MemoryTraits<Kokkos::Unmanaged | Kokkos::RandomAccess>>;
+
+  view_t nonConst = issue_5594::create_view<view_t>();
+  view_const_t view(nonConst);
+  um_view_const_t uview(&view(0), view.size());
+
+  bool err;
+  Kokkos::parallel_reduce(
+      "test_view_subview_const_randomaccess",
+      Kokkos::RangePolicy<execution_space>(0, view.extent(0)),
+      KOKKOS_LAMBDA(const int i, bool& lerr) { lerr = view(i) != uview(i); },
```

Comment:
Not all configurations for all backends support device lambdas. Please just refactor to use a separate functor class.